main.cpp

```diff
@@ -115,17 +120,18 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T bucketA( 5, "Bucket A" );                                             //6
+    T bucketB( 7, "Bucket B" );                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    F f;                                            //7
+    auto* smaller = f.compare( &bucketA, &bucketB );                              //8
+    if (smaller != nullptr)
+		std::cout << "the smaller one is << " << smaller->name << std::endl; //9
```

Comment:
Fix this indentation
Add an `else` block that prints out why `smaller` is `nullptr`
- [ ] revise